Beams/Widgets/qMRMLBeamParametersTabWidget.cxx

```diff
@@ -652,6 +713,117 @@ void qMRMLBeamParametersTabWidget::mlcPositionDoubleArrayNodeChanged(vtkMRMLNode
 #endif
 }
 
+//-----------------------------------------------------------------------------
+void qMRMLBeamParametersTabWidget::generateMLCboundaryClicked()
+{
+  Q_D(qMRMLBeamParametersTabWidget);
+
+  if (!d->mlcPositionLogic)
+  {
+    qCritical() << Q_FUNC_INFO << ": MLC position calculation logic is invalid!";
+    return;
+  }
+
+  bool mlcType = d->radioButton_MLCX->isChecked();
+  int nofPairs = static_cast<int>(d->SliderWidget_NumberOfLeafPairs->value());
+  double leafPairSize = d->SliderWidget_LeafPairBoundarySize->value();
+  double offset = d->SliderWidget_IsocenterOffset->value();
+  
+  vtkMRMLTableNode* mlcTable = d->mlcPositionLogic->CreateMultiLeafCollimatorTableNodeBoundaryData( 
+    mlcType, nofPairs, leafPairSize, offset);
+  if (!mlcTable)
+  {
+    qCritical() << Q_FUNC_INFO << ": Unable to create MLC boundary data table!";
+  }
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLBeamParametersTabWidget::updateMLCboundaryClicked()
+{
+  Q_D(qMRMLBeamParametersTabWidget);
+
+  if (!d->mlcPositionLogic)
+  {
+    qCritical() << Q_FUNC_INFO << ": MLC position calculation logic is invalid!";
+    return;
+  }
+
+  bool mlcType = d->radioButton_MLCX->isChecked();
+  int nofPairs = static_cast<int>(d->SliderWidget_NumberOfLeafPairs->value());
+  double leafPairSize = d->SliderWidget_LeafPairBoundarySize->value();
+  double offset = d->SliderWidget_IsocenterOffset->value();
+
+  vtkMRMLNode* node = d->MRMLNodeComboBox_MLCBoundaryAndPositionTable->currentNode();
+  vtkMRMLTableNode* mlcTableNode = vtkMRMLTableNode::SafeDownCast(node);
+  if (mlcTableNode && d->radioButton_MLCX->isChecked())
+  {
+    d->mlcPositionLogic->UpdateMultiLeafCollimatorTableNodeBoundaryData( mlcTableNode,
+    mlcType, nofPairs, leafPairSize, offset);
+  }
+  else
+  {
+    qCritical() << Q_FUNC_INFO << ": Unable to update MLC boundary data table!";
+  }
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLBeamParametersTabWidget::calculateMLCPositionClicked()
+{
+  Q_D(qMRMLBeamParametersTabWidget);
+  vtkMRMLNode* mlcTable = d->MRMLNodeComboBox_MLCBoundaryAndPositionTable->currentNode();
+
+  if (!d->BeamNode)
+  {
+    qCritical() << Q_FUNC_INFO << ": No current beam node!";
+    return;
+  }
+  vtkMRMLRTPlanNode* planNode = d->BeamNode->GetParentPlanNode();
+  if (!planNode)
+  {
+    qCritical() << Q_FUNC_INFO << ": No plan node for current beam!";
+    return;
+  }
+  vtkMRMLSegmentationNode* segmentationNode = planNode->GetSegmentationNode();
+  vtkMRMLScalarVolumeNode* volumeNode = planNode->GetReferenceVolumeNode();
+  const char* targetID = planNode->GetTargetSegmentID();
+
+  if (!segmentationNode || !volumeNode || !targetID)
+  {
+    qCritical() << Q_FUNC_INFO << ": No segmentation, reference volume or target id data!";
+    return;
+  }
+
+  qDebug() << Q_FUNC_INFO << "Target ID: " << targetID;

```

Comment:
Please remove if not necessary for some reason. And if so, please add ": " before "Target ID".

Reply:
Removed.